web/src/HelpSearchBar.tsx

```diff
@@ -0,0 +1,114 @@
+import { InputAdornment } from "@material-ui/core"
+import { InputProps as StandardInputProps } from "@material-ui/core/Input/Input"
+import React, { ChangeEvent, KeyboardEvent, useState } from "react"
+import styled from "styled-components"
+import { ReactComponent as CloseSvg } from "./assets/svg/close.svg"
+import { ReactComponent as SearchSvg } from "./assets/svg/search.svg"
+import {
+  InstrumentedButton,
+  InstrumentedTextField,
+} from "./instrumentedComponents"
+import {
+  Color,
+  Font,
+  FontSize,
+  mixinResetButtonStyle,
+  SizeUnit,
+} from "./style-helpers"
+
+export function searchDocs(query: string) {
+  window.open(
+    `https://docs.tilt.dev/search?q=${encodeURI(query)}&utm_source=tiltui`
+  )
+}
+
+export const HelpSearchBarTextField = styled(InstrumentedTextField)`
+  & .MuiOutlinedInput-root {
```

Comment:
(optional design suggestion) wdyt about giving the text field a `100%` width? that way, a user has more room to see their text input if it's a little longer. cc: @SurbhiSGupta 

![Screen Shot 2022-01-18 at 12 05 22 PM](https://user-images.githubusercontent.com/23283986/149984971-58f85c9c-6d66-4ce6-817c-b5fa3ca34bef.png)

Reply:
Visually the close search bar icon X may be too close to the dialogue close icon. Actual interaction wise, they may not conflict at all !

Reply:
I will leave it as is, then.

Reply:
I will leave this as is, then.